### PR TITLE
fix: Extract Function produces duplicate fn names

### LINKED ISSUE #10037: Extract to function assist creates duplicate function definitions
The 'extract to function' assist currently always generates a function or method named `fun_name` AFAICT, even if such a function or method is already in scope.

### PR DESCRIPTION
**Pull request by DorianListens**
This change fixes #10037, in more or less the most naive fashion
possible.

We continue to start with the hardcoded default of "fun_name", and now append a
counter to the end of it if that name is already in scope.

In the future, we can probably apply more heuristics here to wind up with more
useful names by default, but for now this resolves the immediate problem.

### CODE REVIEW

File: crates/ide-assists/src/handlers/extract_function.rs [hunk 1]
```diff
@@ -91,6 +92,8 @@ pub(crate) fn extract_function(acc: &mut Assists, ctx: &AssistContext) -> Option
 
     let scope = ImportScope::find_insert_use_container(&node, &ctx.sema)?;
 
+    let name = make_function_name(&semantics_scope);
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/extract_function.rs:95]: This can be moved into the builder closure

Comment (DorianListens) [reply crates/ide-assists/src/handlers/extract_function.rs]: Done!

**Comment by DorianListens** [comment]
Thanks @Veykril 

@bors r+

**Pull request merged**